src/core/exec-invoke.c

```diff
@@ -4592,6 +4596,16 @@ int exec_invoke(
                                                             "Failed to adjust ownership of '%s', ignoring: %m", memory_pressure_path);
                                         memory_pressure_path = mfree(memory_pressure_path);
                                 }
+                                /* First we use the current cgroup path to chmod and chown the memory pressure path, then pass the path relative
+                                 * to the cgroup namespace to environment variables and mounts. */
+                                if (memory_pressure_path && needs_sandboxing && exec_needs_cgroup_namespace(context, params)) {
```

Comment:
the conditionalization left of the first && seems unnecessary?

Reply:
Well...we technically need it because we could hit line 4597 aka
```
memory_pressure_path = mfree(memory_pressure_path);
```

if we fail to chown/chmod memory pressure path. In that case, we should probably not set memory pressure path environment variable, otherwise, the unit may not be able to read it.

I'll add a note why we have this conditional

Reply:
oh right